### PR TITLE
UI: Fix empty item on kv list

### PR DESCRIPTION
**Pull request by hashishaw**
There is a bug in our version of ember-data (4.11.x) where after editing a record, going to the list page would result in a ghost record because `unloadAll` wasn't unloading everything properly. As part of the investigation I found a couple other issues that I cleaned up as well. 

**Replication steps for initial issue:**
1. Have a few secrets created in a KV v2 engine
2. Go to a secret detail
3. Click "Metadata" tab
4. Edit metadata, add custom metadata
5. Save
6. Click the breadcrumb to go back to the secrets list 
7. Ghost item exists on the list

We decided not to upgrade to ember-data@4.12 at this time (the version which has a fix), because after upgrade other tests were failing with `expected stable identifier` issues, and the upgrade requires store configuration which we didn't want to rush to get in before code freeze without adequate testing.

### CODE REVIEW

File: ui/app/services/store.js [hunk 1]
```diff
@@ -138,13 +138,23 @@ export default class StoreService extends Store {
     return resp;
   }
 
+  forceUnload(modelName) {
```
Comment (Monkeychip) [inline ui/app/services/store.js:141]: interesting find!

File: ui/lib/kv/addon/routes/list-directory.js [hunk 2]
```diff
@@ -46,9 +46,15 @@ export default class KvSecretsListRoute extends Route {
       });
   }
 
+  getPathToSecret(pathParam) {
```
Comment (Monkeychip) [inline ui/lib/kv/addon/routes/list-directory.js:49]: much cleaner, thank you!

File: ui/app/services/store.js [hunk 3]
```diff
@@ -138,13 +138,23 @@ export default class StoreService extends Store {
...
+    // For some reason peeking the records before unloading ensures we
+    // don't have ghost models leftover
+    // eslint-disable-next-line no-console
+    console.info(`unloading ${this.peekAll(modelName).length} records`);
+    run(() => this.unloadAll(modelName));
+    return;
```
Comment (zofskeez) [inline ui/app/services/store.js:147]: Nit but this return seems unnecessary since it will return at the end of the function anyway.

File: ui/app/services/store.js
Comment (zofskeez) [inline ui/app/services/store.js:1]: Very interesting situation here and feels like an Ember Data bug. Either that or we're doing something somewhere that is not obvious that might be causing those specific records to be bypassed by `unloadAll`.

In any case, if this solves the issue for the time being it looks good to me. The only suggestion I might have is rather than logging out the result of `peekAll`, loop the array and call `unloadRecord` on each. I think that is more or less what `unloadAll` does behind the scenes. My thinking is that in case there is perhaps an issue with `unloadAll` you can be sure that every record returned by the peek is unloaded.

Comment (hashishaw) [reply ui/app/services/store.js:1]: I think the issue with ember-data is somewhere upstream, because I tried unloading each after peekAll and one of the records we attempt to unload is undefined (even though in the peekAll list, it's fine) 😵 so bizarre

File: ui/app/components/secret-delete-menu.js [hunk 4]
```diff
@@ -24,92 +22,21 @@ export default class SecretDeleteMenu extends Component {
 
   @tracked showDeleteModal = false;
 
-  @maybeQueryRecord(
```
Comment (hashishaw) [inline ui/app/components/secret-delete-menu.js:27]: When navigating from the details page back to the list, the console was erroring with 
```Cannot create a new tag for `<model::capabilities:undefined>` after it has been destroyed```
Finally tracked it down to these not tearing down gracefully, so I moved the capabilities to the model instead which is our latest pattern

File: ui/app/components/secret-delete-menu.js [hunk 5]
```diff
@@ -24,92 +22,21 @@ export default class SecretDeleteMenu extends Component {
...
-      };
-    },
-    'isV2',
-    'model',
-    'model.id',
-    'mode'
-  )
-  secretSoftDataPath;
-  @alias('secretSoftDataPath.canUpdate') canSoftDeleteSecretData;
+  get canUndeleteVersion() {
```
Comment (hashishaw) [inline ui/app/components/secret-delete-menu.js:25]: This component should actually go away soon, so to simplify the changeset I just made these all getters of the same name

File: ui/app/components/secret-delete-menu.js (hunk 4 above)
Comment (zofskeez) [reply ui/app/components/secret-delete-menu.js:27]: It looks like the `secret-edit` component also uses `maybeQueryRecord` so something to be aware of if we notice that error further. Interesting that it's also used in `lazy-capabilities` which is used a bunch but maybe it's ok when used on a model since they aren't torn down as frequently as components.

File: ui/app/services/store.js [hunk 6]
```diff
@@ -138,12 +138,22 @@ export default class StoreService extends Store {
...
+    run(() => this.unloadAll(modelName));
+  }
+
   // pushes records into the store and returns the result
   fetchPage(modelName, query) {
     const response = this.constructResponse(modelName, query);
-    this.unloadAll(modelName);
+    this.forceUnload(modelName);
+    // Hack to ensure the pushed records below all get in the store
+    this.peekAll(modelName).length;
```
Comment (zofskeez) [inline ui/app/services/store.js:154]: I'm guessing this is necessary but since there is already a peek in the `forceUnload` method it has to be done again after `unloadAll`?

File: ui/app/services/store.js
Comment (zofskeez) [inline ui/app/services/store.js:1]: Just a thought but I wonder if someone else ends up doing the next upgrade if it will be clear what can be removed as part of these workarounds? Maybe some extra annotations like _remove after upgrade to Ember Data 4.12_ would be helpful so we don't have extra peeks etc. hanging around.

File: ui/tests/acceptance/secrets/backend/kv/kv-v2-workflow-edge-cases-test.js [hunk 7]
```diff
@@ -249,6 +251,19 @@ module('Acceptance | kv-v2 workflow | edge cases', function (hooks) {
...
+    assert.dom(PAGE.list.item()).exists({ count: 2 }, 'two secrets are listed');
+    await click(PAGE.list.item('two'));
+    await click(PAGE.secretTab('Metadata'));
+    await click(PAGE.metadata.editBtn);
+    await fillIn(FORM.keyInput(), 'foo');
+    await fillIn(FORM.valueInput(), 'bar');
+    await click(FORM.saveBtn);
+    await click(PAGE.breadcrumbAtIdx(2));
+    assert.dom(PAGE.list.item()).exists({ count: 2 }, 'two secrets are listed');
+  });
```
Comment (zofskeez) [inline ui/tests/acceptance/secrets/backend/kv/kv-v2-workflow-edge-cases-test.js:266]: 🎉

**Review by zofskeez — APPROVED** [review]
Great working grinding this one out and finding a work around 👏. I left a few comments and questions, nothing blocking.

File: ui/app/services/store.js
Comment (hashishaw) [reply ui/app/services/store.js:1]: Good idea! Added comments about removing once we upgrade to 4.12
